framework/src/vectorpostprocessors/VolumeHistogram.C

```diff
@@ -0,0 +1,93 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "VolumeHistogram.h"
+
+// libmesh includes
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<VolumeHistogram>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+  params.addParam<unsigned int>("bin_number", 50, "Number of histogram bins");
+  params.addCoupledVar("variable", "Variable to bin the volume of");
+  params.addRequiredParam<Real>("min_value", "Minimum variable value");
+  params.addRequiredParam<Real>("max_value", "Maximum variable value");
+  return params;
+}
+
+VolumeHistogram::VolumeHistogram(const InputParameters & parameters) :
+    ElementVectorPostprocessor(parameters),
+    _nbins(getParam<unsigned int>("bin_number")),
+    _min_value(getParam<Real>("min_value")),
+    _max_value(getParam<Real>("max_value")),
+    _deltaV((_max_value - _min_value) / _nbins),
+    _value(coupledValue("variable")),
+    _bin_center(declareVector(getVar("variable", 0)->name())),
```

Comment:
It is, it's just that if you plan to couple to this vector having the name change out from underneath you will be a bit of a pain. If that's not your use case, no problem. Just a suggestion.